src/index.js

```diff
@@ -136,17 +207,107 @@ module.exports = {
+      console.log('globalMeanY is :', globalMeanY);
       /* eslint-enable no-console */
+ 
+      // Extract X and y variable name list from userData
+      const y_label=opts.userResults[0].userData.__FEATURES__;
+      const X_label=Object.keys(opts.userResults[0].userData.files[0].tags);
+      console.log('X_label is', X_label);
+      console.log('y_label is',y_label);
 
       return {
         averageBetaVector,
+        globalMeanY,
+        X_label,
+        y_label,
+//        complete: true,
+      };
+    },
+    verbose: true,
+  }, {
+    // get sstNode from each local node and calculate the statistics
+    type: 'function',
+    fn(opts) {
+
+    const userResults = opts.userResults;
+    const X_label=opts.previousData.X_label;
+    const y_label=opts.previousData.y_label;
+
+    // get passed parameters from local nodes
+    const averageBetaVector = userResults[0].data.averageBetaVector;
+    const betaVectorLocal = userResults.map(r => r.data.betaVector);
+    const rSquaredLocalOriginal = userResults.map(r => r.data.rSquared);
+    const tValueLocalOriginal = userResults.map(r => r.data.tValue);
+    const pValueLocalOriginal = userResults.map(r => r.data.pValue);
+    const rSquaredLocal = userResults.map(r => r.data.rSquaredLocal);
+    const tValueLocal = userResults.map(r => r.data.tValueLocal);
+    const pValueLocal = userResults.map(r => r.data.pValueLocal);
+
+    //calculate global parameters
+    const sseGlobal = userResults.reduce((sum, userResult) => sum + userResult.data.sseLocal, 0);
+    const sstGlobal = userResults.reduce((sum, userResult) => sum + userResult.data.sstLocal, 0);
+    const globalYCount = userResults.reduce((sum,userResult) => sum + userResult.data.localCount,0);
+    const betaCount = userResults[0].data.averageBetaVector.length;
+    const varError = (1 /(globalYCount-2)) * sseGlobal;
+    const varXGlobal = [];   
+    const seBetaGlobal = [];
+    const tValueGlobal = [];
+
+   // calculate tValueGlobal 
+    for (let i = 0; i < betaCount; i += 1) {
+        varXGlobal[i] = userResults.reduce((sum, userResult) => sum + userResult.data.varXLocal[i], 0 );
+        seBetaGlobal[i] = Math.sqrt(varError/varXGlobal[i]);
+        tValueGlobal[i] = averageBetaVector[i]/seBetaGlobal[i];
+       }
+
+  // calculate r squared global    
+    const rSquaredGlobal = 1-(sseGlobal/sstGlobal);
+    
+  // add t to p value transformation //
+    var tdist=distributions.Studentt(globalYCount-1);
+    const tcdf = tValueGlobal.map(r => tdist.cdf(r));
+    const pValueGlobal=n.mul(2,(n.sub(1,tcdf))); // two tail pValue
+
+    console.log('The global r squared for averageBetaVector :', rSquaredGlobal);
+    console.log('The global t Values for averageBetaVector :', tValueGlobal);
+    console.log('The global p Values for averageBetaVector :', pValueGlobal);
+
+  // now each returned variable is full-precision floating number, may need reduce the precision when disply on the table 
```

Comment:
This comment isn’t necessary: it doesn’t add value to the code. Please use a [docblock with `@returns`](http://usejsdoc.org/tags-returns.html).